core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -2359,7 +2359,7 @@ class KafkaController(val config: KafkaConfig,
           } else if (newLeaderAndIsr.partitionEpoch < currentLeaderAndIsr.partitionEpoch) {
             partitionResponses(tp) = Left(Errors.INVALID_UPDATE_VERSION)
             None
-          }  else if (newLeaderAndIsr.leaderRecoveryState == LeaderRecoveryState.RECOVERING && newLeaderAndIsr.isr.length > 1) {
+          }  else if (newLeaderAndIsr.leaderRecoveryState == LeaderRecoveryState.RECOVERING && newLeaderAndIsr.isr.lengthCompare(1) > 0) {
```

Comment:
FYI reviewers -
`newLeaderAndIsr.isr` is a scala `List`, hence, lengthCompare() is beneficial.